ui/app/AppLayouts/Communities/controls/PermissionsRow.qml

```diff
@@ -112,8 +112,16 @@ Control {
 
         property bool dotsVisible: false
 
+        function checkPermissionVisible(permission) {
+            if (permission.isPrivate) {
+                return permission.tokenCriteriaMet
+            }
+            return true
+        }
+
         function buildShortModel(model) {
```

Comment:
The model count gives all the entries, including the ones that should be ultimately hidden as per the component's logic. I don't see much difference between filtering like this in one pass, or in 2 passes? Am I missing something?

Reply:
OTOH, I could probably pre-filter the input model using a SFPM

Reply:
The layout is dependent on number of visible entries, not entries in total.
Imagine that we have 5 entries, 3 are visible and they met that:

```qml
                if((holdingsCount1 === 1 && holdingsCount2 === 1 && holdingsCount3 === 1) ||
                   (holdingsCount1 === 2 && holdingsCount2 === 1 && holdingsCount3 === 1) ||
                   (holdingsCount1 === 1 && holdingsCount2 === 2 && holdingsCount3 === 1) ||
                   (holdingsCount1 === 1 && holdingsCount2 === 1 && holdingsCount3 === 2)) {
```
It means that 3 groups should be displayed.

But we will end up with just 2 groups (case 3) because the if above is not even checked as it's guarded by

```qml
if(modelCount === 3) {
```
which is not met as it doesn't (but should) check number of visible entries.

Reply:
It would be cool to have unit test covering this quite complex and error-prone logic :)

Reply:
@micieslak done! pls check it out

Reply:
Ahh, I posted explanation in the meantime. Not sure if SFPM is a bit an overkill here, but for sure it's valid now.

Reply:
> The layout is dependent on number of visible entries, not entries in total. Imagine that we have 5 entries, 3 are visible and they met that:
> 
> ```qml
>                 if((holdingsCount1 === 1 && holdingsCount2 === 1 && holdingsCount3 === 1) ||
>                    (holdingsCount1 === 2 && holdingsCount2 === 1 && holdingsCount3 === 1) ||
>                    (holdingsCount1 === 1 && holdingsCount2 === 2 && holdingsCount3 === 1) ||
>                    (holdingsCount1 === 1 && holdingsCount2 === 1 && holdingsCount3 === 2)) {
> ```
> 
> It means that 3 groups should be displayed.
> 
> But we will end up with just 2 groups (case 3) because the if above is not even checked as it's guarded by
> 
> ```qml
> if(modelCount === 3) {
> ```
> 
> which is not met as it doesn't (but should) check number of visible entries.

Yeah I realized that and it's fixed now :)